setup.py

```diff
@@ -881,205 +791,46 @@ def run(self):
     else:
         cuda_lib_dirs = ['lib64', 'lib']
-
         for lib_dir in cuda_lib_dirs:
             cuda_lib_path = os.path.join(CUDA_HOME, lib_dir)
             if os.path.exists(cuda_lib_path):
                 break
-        extra_link_args.append('-Wl,-rpath,' + cuda_lib_path)
-
-        nvtoolext_lib_name = 'nvToolsExt'
-
     library_dirs.append(cuda_lib_path)
-    cuda_include_path = os.path.join(CUDA_HOME, 'include')
-    include_dirs.append(cuda_include_path)
-    include_dirs.append(tmp_install_path + "/include/THCUNN")
-    extra_compile_args += ['-DUSE_CUDA']
-    extra_compile_args += ['-DCUDA_LIB_PATH=' + cuda_lib_path]
-    main_libraries += ['cudart', nvtoolext_lib_name]
-    main_sources += [
-        "torch/csrc/cuda/Module.cpp",
-        "torch/csrc/cuda/Storage.cpp",
-        "torch/csrc/cuda/Stream.cpp",
-        "torch/csrc/cuda/utils.cpp",
-        "torch/csrc/cuda/comm.cpp",
-        "torch/csrc/cuda/python_comm.cpp",
-        "torch/csrc/cuda/serialization.cpp",
-        "torch/csrc/nn/THCUNN.cpp",
-    ]
-
-if USE_ROCM:
-    rocm_include_path = '/opt/rocm/include'
-    hcc_include_path = '/opt/rocm/hcc/include'
-    rocblas_include_path = '/opt/rocm/rocblas/include'
-    hipsparse_include_path = '/opt/rocm/hipsparse/include'
-    rocfft_include_path = '/opt/rocm/rocfft/include'
-    hiprand_include_path = '/opt/rocm/hiprand/include'
-    rocrand_include_path = '/opt/rocm/rocrand/include'
-    thrust_include_path = '/opt/rocm/include/'
-    hip_lib_path = '/opt/rocm/hip/lib'
-    hcc_lib_path = '/opt/rocm/hcc/lib'
-    include_dirs.append(rocm_include_path)
-    include_dirs.append(hcc_include_path)
-    include_dirs.append(rocblas_include_path)
-    include_dirs.append(rocfft_include_path)
-    include_dirs.append(hipsparse_include_path)
-    include_dirs.append(hiprand_include_path)
-    include_dirs.append(rocrand_include_path)
-    include_dirs.append(thrust_include_path)
-    include_dirs.append(tmp_install_path + "/include/THCUNN")
-    extra_link_args.append('-L' + hip_lib_path)
-    extra_link_args.append('-Wl,-rpath,' + hip_lib_path)
-    extra_compile_args += ['-DUSE_ROCM']
-    extra_compile_args += ['-D__HIP_PLATFORM_HCC__']
-
-    main_sources += [
-        "torch/csrc/cuda/Module.cpp",
-        "torch/csrc/cuda/Storage.cpp",
-        "torch/csrc/cuda/Stream.cpp",
-        "torch/csrc/cuda/utils.cpp",
-        "torch/csrc/cuda/comm.cpp",
-        "torch/csrc/cuda/python_comm.cpp",
-        "torch/csrc/cuda/serialization.cpp",
-        "torch/csrc/nn/THCUNN.cpp",
-    ]
-
-if USE_NCCL:
-    if USE_SYSTEM_NCCL:
-        include_dirs.append(NCCL_INCLUDE_DIR)
-    else:
-        include_dirs.append("build/nccl/include")
-    extra_compile_args += ['-DUSE_NCCL']
-    main_sources += [
-        "torch/csrc/cuda/nccl.cpp",
-        "torch/csrc/cuda/python_nccl.cpp",
-    ]
-if USE_CUDNN:
-    main_libraries += [CUDNN_LIBRARY]
-    # NOTE: these are at the front, in case there's another cuDNN in CUDA path
```

Comment:
double-checking: is this subtlety handled on the CMake side, can you please check on this @anderspapitto 

Basically, this is the case where $CUDA_HOME/lib64 has an old or incompatible `libcudnn.so`, which we can inadvertently link to if we're not careful

Reply:
handled it below in accordance with your comment there